content/scripts/build.js

```diff
@@ -982,7 +982,19 @@ class Builder {
     const rawHtml = fs.readFileSync(path.join(folder, "index.html"), "utf8");
     hasher.update(rawHtml);
 
-    const [renderedHtml, errors] = await this.renderHtml(rawHtml, metadata);
+    let renderedHtml;
+
+    // When the source has this set to truthy it simple means, the 'index.html'
+    // is already fully rendered HTML.
+    if (source.htmlAlreadyRendered) {
+      renderedHtml = rawHtml;
+    } else {
+      const renderingResult = await this.renderHtml(rawHtml, metadata);
+      renderedHtml = renderingResult[0];
+      // XXX The this.renderHtml() method actually returns a tuple
```

Comment:
This is something we'll dig into deeper when we try to merge `ks-renderer` into `master`. That and the fact that individual exceptions thrown doesn't appear to quit the builder immediately.

Reply:
I like
```javascript
const [renderedHtml, errors] = await this.renderHtml(rawHtml, metadata);
```
better than
```javascript
const renderingResult = await this.renderHtml(rawHtml, metadata);
renderedHtml = renderingResult[0];
```
but perhaps you changed it because you didn't like the unused `errors`? Or perhaps you think we should be throwing exceptions on KS rendering errors (and not returning a list of them), and immediately stopping the entire build instead?

Reply:
It's because you can't do this:
```javascript
if (condition()) {
  const renderedHtml = "foo"
} else {
  const [renderedHtml, errors] = get()
}
```
That's `ReferenceError`. 

But perhaps besides the point. You can do:
```javascript
let renderedHtml
if (condition()) {
  renderedHtml = "foo"
} else {
  [renderedHtml, errors] = get()
}
```
But still leaves you with an unused variable which I think is ugly and breaks `eslint`.
However, in typing up this comment I discovered a way that works and doesn't break `eslint`:

```javascript
let renderedHtml;
if (condition()) {
  renderedHtml = "foo";
} else {
  [renderedHtml] = get();
}
```
See
```
> var LIST = ['one', 'two', 'three']
> var [first] = LIST
> first
'one'
```

Reply:
Ah, I see. I also learned something new, as I didn't even realize that you could do `var [first] = LIST`.